test/one/profiler/test/Runner.java

```diff
@@ -0,0 +1,242 @@
+            libExt = "dylib";
+            return Os.MACOS;
+        } else if (osName.contains("windows")) {
+            libExt = "dll";
+            return Os.WINDOWS;
+        }
+        throw new IllegalStateException("Unknown OS type");
+    }
+
+    private static int detectJvmVersion() { // Not sure if this works for zing
+        String prop = System.getProperty("java.vm.version");
+
+        if (prop.startsWith("25.") && prop.charAt(3) > '0') {
+            return 8;
+        } else if (prop.startsWith("24.") && prop.charAt(3) > '0') {
+            return 7;
+        } else if (prop.startsWith("20.") && prop.charAt(3) > '0') {
+            return 6;
+        } else {
+            String substring = prop.substring(0, prop.indexOf('.'));
+            int possibleVersion = Integer.parseInt(substring);
+            if (possibleVersion < 9) {
+                return 9;
+            }
+            return possibleVersion;
+        }
+    }
+
+    private static Jvm detectJvm() throws RuntimeException{
+        // Example javaHome: /usr/lib/jvm/amazon-corretto-17.0.8.7.1-linux-x64
+        File javaHome = new File(System.getProperty("java.home"));
+
+        // J9 Check which relies on JDK 8 JRE file dir differences
+        File libDirectory = new File(javaHome, "lib");
+        Pattern j9IndicatorFilePattern = Pattern.compile("J9TraceFormat.dat");
+        File[] files = libDirectory.listFiles();
+        if (files != null) {
+            for (File file : files) {
+                if (j9IndicatorFilePattern.matcher(file.getName()).matches()) {
+                    return Jvm.J9;
+                }
+            }
+        }
+
+        // Zing check
+
+        // Workaround for java.home including /jre in JDK 8
+        if (currentJvmVersion == 8) {
+            javaHome = javaHome.getParentFile();
+        }
+
+        // Workaround for contents/home on mac
+        if (currentOs == Os.MACOS) {
+            javaHome = javaHome.getParentFile();
+        }
+        File etcDirectory = new File(javaHome, "etc");
+        File zing = new File(etcDirectory, "zing");
+        if (zing.exists()) {
+            return Jvm.ZING;
+        }
+
+        // Otherwise it's some variation of hotspot
+        return Jvm.HOTSPOT;
+    }
+
+    private static Arch detectArch() throws RuntimeException{
+        String architecture = System.getProperty("os.arch");
+
+        if (architecture.contains("x86_64") || architecture.contains("amd64")) {
+            return Arch.X64;
+        } else if (architecture.contains("aarch64")) {
+            return Arch.ARM64;
+        } else if (architecture.contains("arm")) {
+            return Arch.ARM32;
+        } else if (architecture.contains("ppc64le")) {
+            return Arch.PPC64LE;
+        } else if (architecture.endsWith("86")) {
+            return Arch.X86;
+        } else {
+            throw new RuntimeException("Unable to detect the OS used by the host.\nOS may be unsupported.");
```

Comment:
Multiline exception message is discouraged.
Also, it's about the architecture, not the OS.

Reply:
Addressed